extensions/positron-notebook-controllers/src/notebookControllerManager.ts

```diff
@@ -46,11 +46,12 @@ export class NotebookControllerManager implements vscode.Disposable {
 	 * @returns Promise that resolves when the notebook's affinity has been updated across all controllers.
 	 */
 	public async updateNotebookAffinity(notebook: vscode.NotebookDocument): Promise<void> {
-		if (notebook.uri.scheme === 'untitled') {
-			// If the notebook is untitled, wait for its data to be updated. This works around the fact
-			// that `vscode.openNotebookDocument(notebookType, content)` first creates a notebook
-			// (triggering `onDidOpenNotebookDocument`), and later updates its content (triggering
-			// `onDidChangeNotebookDocument`).
+		const cells = notebook.getCells();
```

Comment:
This is the actual fix. We had previously implemented it for `untitled` schemes, but it needs to apply to `file` schemes too if the notebook is empty. 

Changes in other files are to give less surprising behavior if we encounter this issue again.